programs/solana/programs/ics26-router/src/instructions/ack_packet.rs

```diff
@@ -56,6 +58,16 @@ pub struct AckPacket<'info> {
 
     /// CHECK: Consensus state account, owned by light client program
     pub consensus_state: AccountInfo<'info>,
+
+    // IBC app accounts for CPI
+    /// CHECK: IBC app program, validated against `IBCApp` account
+    #[account(
+        constraint = ibc_app_program.key() == ibc_app.app_program_id @ RouterError::IbcAppNotFound
+    )]
+    pub ibc_app_program: AccountInfo<'info>,
+
+    /// CHECK: IBC app state account, owned by IBC app program
+    pub ibc_app_state: AccountInfo<'info>,
```

Comment:
So every IBC app needs to have exactly 1 `app_state` account for these entry points?